lib/configure.js

```diff
@@ -360,3 +278,97 @@ function configure (gyp, argv, callback) {
   }
 
 }
+
+function findPython (python, callback) {
+  checkPython()
+
+  // Check if Python is in the $PATH
+  function checkPython () {
+    log.verbose('check python', 'checking for Python executable "%s" in the PATH', python)
+    which(python, function (err, execPath) {
+      if (err) {
+        log.verbose('`which` failed', python, err)
+        if (python === 'python2') {
+          python = 'python'
+          return checkPython()
+        }
+        if (win) {
+          guessPython()
+        } else {
+          failNoPython()
+        }
+      } else {
+        log.verbose('`which` succeeded', python, execPath)
+        checkPythonVersion()
+      }
+    })
+  }
+
+  // Called on Windows when "python" isn't available in the current $PATH.
+  // We're gonna check if "%SystemDrive%\python27\python.exe" exists.
+  function guessPython () {
+    log.verbose('could not find "' + python + '". guessing location')
+    var rootDir = process.env.SystemDrive || 'C:\\'
+    if (rootDir[rootDir.length - 1] !== '\\') {
+      rootDir += '\\'
+    }
+    var pythonPath = path.resolve(rootDir, 'Python27', 'python.exe')
+    log.verbose('ensuring that file exists:', pythonPath)
+    fs.stat(pythonPath, function (err, stat) {
+      if (err) {
+        if (err.code == 'ENOENT') {
+          failNoPython()
+        } else {
+          callback(err)
+        }
+        return
+      }
+      python = pythonPath
+      checkPythonVersion()
+    })
+  }
+
+  function checkPythonVersion () {
+    var env = extend({}, process.env)
+    env.TERM = 'dumb'
+
+    execFile(python, ['-c', 'import platform; print(platform.python_version());'], { env: env }, function (err, stdout) {
+      if (err) {
+        return callback(err)
+      }
+      log.verbose('check python version', '`%s -c "import platform; print(platform.python_version());"` returned: %j', python, stdout)
+      var version = stdout.trim()
+      if (~version.indexOf('+')) {
+        log.silly('stripping "+" sign(s) from version')
+        version = version.replace(/\+/g, '')
+      }
+      if (~version.indexOf('rc')) {
+        log.silly('stripping "rc" identifier from version')
+        version = version.replace(/rc(.*)$/ig, '')
```

Comment:
I agree it's quirky but please keep in mind that I only moved this code around, I didn't edit any of it.